pkg/apis/networking/v1/defaults.go

```diff
@@ -43,3 +47,9 @@ func SetDefaults_NetworkPolicy(obj *networkingv1.NetworkPolicy) {
 		}
 	}
 }
+
+func SetDefaults_HTTPIngressPath(obj *networkingv1.HTTPIngressPath) {
+	if obj.PathType == nil {
+		obj.PathType = &defaultPathType
```

Comment:
Although I'd initially been working towards making pathType a required field with no default, we ended up deciding against that to avoid making such a significant change as part of a v1 release. This would just make it more work for end users to move to the v1 API - this field would need to be set multiple times (1 per path) for each Ingress. 

I was also hesitant to drag out the Ingress v1 graduation process as the work on Service APIs was gaining momentum. As I understand it, requiring the pathType field would require another beta release in 1.18 before we could reach v1 in 1.19. I think there's relatively minimal value gained by that approach. For more context:

https://github.com/kubernetes/enhancements/pull/1413#issuecomment-571360190

Reply:
> This would just make it more work for end users to move to the v1 API - this field would need to be set multiple times (1 per path) for each Ingress.

Propagating an ambiguous meaning of path as a default to v1 doesn't seem great.

> As I understand it, requiring the pathType field would require another beta release in 1.18 before we could reach v1 in 1.19.

If the v1beta1 versions default it, and v1 does not, then validation can require it without affecting v1beta1 API users.